ClientApp/viking-vault/src/UserDashboard/BlockedServices.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import {constants} from "../Resources/Constants";
+import SideBar from "../Common/SideBar";
+import TopBar from "../Common/TopBar";
+import UserIcon from "../Common/UserIcon";
+import "../Common/styles.css";
+
+interface IBlockedServicesState {
+    userHasCard: boolean | null;
+    isCardBlocked: boolean | null;
+}
+
+class BlockedServices extends React.Component<any, IBlockedServicesState> {
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            userHasCard: null,
+            isCardBlocked: null
+        }
+    }
+
+    componentDidMount() {
+        this.checkUserHasCard();
+        this.checkCardIsBlocked();
+    }
+
+    checkUserHasCard = () => {
+        const token = sessionStorage.getItem("Authentication-Token");
+        if(token === null) {
+            return;
+        }
+        fetch(constants.baseUrl+"card", {
+            method: "GET",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token': token
+            }
+        }).then(response => {
+            if(response.status !== 200) {
+                return null;
+            }
+            return response.json();
+        }).then(result => {
+            if(result === null) {
+                return;
+            }
+            this.setState({
+                userHasCard: result
+            });
+        });
+    }
+
+    checkCardIsBlocked = () => {
+        const token = sessionStorage.getItem("Authentication-Token");
+        if(token === null) {
+            return;
+        }
+        fetch(constants.baseUrl+"blockedcard", {
+            method: "GET",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token': token
+            }
+        }).then(response => {
+            if(response.status !== 200) {
+                return null;
+            }
+            return response.json();
+        }).then(result => {
+            if(result === null) {
+                return;
+            }
+            console.log(result);
+            this.setState({
+                isCardBlocked: result
+            });
+        });
+    }
+
+    returnBlockServiceView = () => {
+        let message = "";
+        if(this.state.userHasCard === false) {
```

Comment:
you could use if(!this.state.userHasCard)

Reply:
Not exactly, because userHasCard is initializes with null. Until we get the response from the backend we don't assume anything. Considering null is a falsy value !this.state.userHasCard would evaluate to true and the you-don't-have-a-card message would be displayed until the answer from the server is received